compat-30.el

```diff
@@ -28,5 +28,32 @@
 ;; TODO Update to 30.1 as soon as the Emacs emacs-30 branch version bumped
 (compat-version "30.0.50")
 
+;;;; Defined in subr.el
+
+(compat-defun copy-tree (tree &optional vectors-and-records) ;; <compat-tests:copy-tree>
+  "Handle copying records when optional arg is non-nil."
+  :extended t
+  (declare (side-effect-free error-free))
+  (if (consp tree)
+      (let (result)
+	(while (consp tree)
+	  (let ((newcar (car tree)))
+	    (if (or (consp (car tree))
+                    (and vectors-and-records
+                         (or (vectorp (car tree)) (recordp (car tree)))))
+		(setq newcar (copy-tree (car tree) vectors-and-records)))
+	    (push newcar result))
+	  (setq tree (cdr tree)))
+	(nconc (nreverse result)
+               (if (and vectors-and-records (or (vectorp tree) (recordp tree)))
+                   (copy-tree tree vectors-and-records)
```

Comment:
Instead of `copy-tree` you should call `compat--copy-tree`, since the compatibility definition is not replacing the original definition of `copy-tree`.

Reply:
Good catch. Thanks.